pkg/armhelpers/azureclient.go

```diff
@@ -100,7 +100,14 @@ func NewAzureClientWithDeviceAuth(env azure.Environment, subscriptionID string)
 				return nil, err
 			}
 			graphSpt.Refresh()
-			return getClient(env, subscriptionID, tenantID, armSpt, graphSpt)
+
+			c := getClient(env, subscriptionID, tenantID, armSpt, graphSpt)
+			err = c.ensureProvidersRegistered(subscriptionID)
+			if err != nil {
```

Comment:
can you move this one level up? The goal is to decouple it from any NewAzureClient* functions.

Reply:
By that you will need to expose ensureProvidersRegistered